src/routers/handlers/upload/upload.ts

```diff
@@ -0,0 +1,61 @@
+import { logger } from "../../../utils/logger";
+import { GenericHandler } from "../generic";
+import { env } from "../../../config";
+import { fileIdPlaceholder, servicePathPrefix, uploadedUrl } from "../../../utils/constants/urls";
+import { Request, Response } from "express";
+import { ContextKeys } from "../../../utils/constants/context.keys";
+import { TransactionService } from "../../../services/external/transaction.service";
+
+export class UploadHandler extends GenericHandler {
+    constructor (private transactionService: TransactionService) {
+        super({
+            title: '',
+            backURL: `${servicePathPrefix}`
+        });
+    }
+
+    async execute (req: Request, _res: Response): Promise<string> {
+        logger.info(`GET Request to send fileId call back address`);
+        const transactionId = await this.callTransactionApi(req, _res);
+        req.session?.setExtraData(ContextKeys.TRANSACTION_ID, transactionId);
+        return getFileUploadUrl(req);
+    }
+
+    async callTransactionApi(req: Request, _res: Response): Promise<string | undefined> {
+        const companyName = req.session?.data.signin_info?.company_number;
+        const reference = env.APP_NAME;
```

Comment:
instead of the if below would this work: 
```suggestion
        const reference = env.APP_NAME || throw new Error("APP_NAME variable undefined");
```